pkgs/by-name/sc/scotch/package.nix

```
@@ -1,15 +1,17 @@
 {
-  bison,
-  bzip2,
-  cmake,
-  fetchFromGitLab,
-  flex,
-  gfortran,
   lib,
-  mpi,
   stdenv,
-  zlib,
+  fetchFromGitLab,
+  cmake,
+  gfortran,
+  bison,
+  flex,
+  bzip2,
   xz,
+  zlib,
+  mpi,
+  withPtScotch ? false,
```

```suggestion
  withPtScotch ? !stdenv.hostPlatform.isMusl,
```
The old default was to enable `PtScotch` implicitly. It might make sense to restore the old default, and only override it to false if not required, e.g. for mumps? Though that does mean we'd be building scotch twice on hydra, so i am not sure. Waiting for at least another review from either of @nim65s or @bzizou before merging this for good.

---

mumps/petsc explicitly declare the scotch variant it want to use (mumps have to override scotch explicitly in case mumps-mpi need the ptscoth variant), So they dont care what the default value for withPtScotch is. 

I would not guarantee override options' stablility as scotch is mainly used as a library by mumps/petsc, in such case, i prefer to keep the interface/default as simple as possilbe.

---

> I would not guarantee override options' stablility as scotch is mainly used as a library by mumps/petsc, in such case, i prefer to keep the interface/default as simple as possilbe.

the default is independent though. You are *changing* the default, and downstream users may have expected PtScotch to just be enabled. Not because you are changing the override interface, but because you are effectively *adding* `(lib.cmakeBool "BUILD_PTSCOTCH" false)`, you are changing default behavior.

---

static is always musl. Musl is a strict superset (because glibc insists on dlopen, which is inherently broken on static)

---

Thanks for fix me on static/musl terminology.
Still, i hold me view to keep the default either true or false: i think it's more important to keep scotch consistent with pkgsMusl.scotch than keep scotch consistent over time.